synapse/util/ratelimitutils.py

```diff
@@ -110,6 +111,9 @@ def ratelimit(self) -> "Iterator[defer.Deferred[None]]":
     def _on_enter(self, request_id: object) -> "defer.Deferred[None]":
         time_now = self.clock.time_msec()
 
+        wait_span_scope = start_active_span("ratelimit wait")
+        wait_span_scope.__enter__()
```

Comment:
This can make a big difference in understanding where all of the time went. In the before, there is just a giant gap with no indication of what's taking up the time. In the after, you can see the `ratelimit wait` span 🎉 

Before |  After
--- | ---
<img width="1190" alt="" src="https://user-images.githubusercontent.com/558581/184039932-53b2cb1e-632f-48cb-a178-e6a75d742832.png"> | <img width="1192" alt="" src="https://user-images.githubusercontent.com/558581/184039930-280e9a16-7bd9-406a-85ae-d9d288a98975.png">